src/OSPSuite.Core/Domain/Services/InitialConditionsCreator.cs

```diff
@@ -54,21 +61,31 @@ public InitialConditionsBuildingBlock CreateFrom(SpatialStructure spatialStructu
          return initialConditions;
       }
 
+      private InitialCondition createInitialCondition(IBuildingBlock initialConditionsBuildingBlock, IEntity container, MoleculeBuilder molecule)
+      {
+         var initialCondition = CreateInitialCondition(_entityPathResolver.ObjectPathFor(container), molecule.Name, molecule.Dimension, molecule.DisplayUnit);
+         setInitialCondition(molecule, initialCondition);
+         setInitialConditionFormula(molecule.DefaultStartFormula, initialCondition, initialConditionsBuildingBlock);
+         return initialCondition;
+      }
+
       private void addMoleculesFrom(InitialConditionsBuildingBlock initialConditionsBuildingBlock, IEntity container, IEnumerable<MoleculeBuilder> molecules)
       {
          foreach (var molecule in molecules)
          {
-            var initialCondition = CreateInitialCondition(_objectPathFactory.CreateAbsoluteObjectPath(container), molecule.Name, molecule.Dimension, molecule.DisplayUnit);
-            setInitialCondition(molecule, initialCondition);
-            setInitialConditionFormula(molecule.DefaultStartFormula, initialCondition, initialConditionsBuildingBlock);
-            initialConditionsBuildingBlock.Add(initialCondition);
+            initialConditionsBuildingBlock.Add(createInitialCondition(initialConditionsBuildingBlock, container, molecule));
          }
       }
 
-      private void setInitialConditionFormula(IFormula formula, InitialCondition initialCondition, IBuildingBlock moleculesStartValues)
+      public void CreateForExpressionInContainers(ExpressionProfileBuildingBlock buildingBlock, IReadOnlyList<IContainer> containers, MoleculeBuilder molecule)
```

Comment:
The signature is a bit weird. It is call create but has a side effects on the given parameter. This methods also acts as a factory (creator factory, same stuff) and typically have no side effects. They just create. 

so I would expect the signature to be more  returning an IreadOnly of InitialCondition and the caller is adding ti t the expression profile. Of course there is the formula cache that I am not sure how to deal with

So maybe the method should be renamed instead= >AddToExpressionProfile so that this is clear that is has side effects. 

I would add a comment explaining also what the contaienrs are because this is not clear